File: Ahorn/entities/ResetStateCrystal.jl
module CommunalHelperResetCrystal

using ..Ahorn, Maple

@mapdef Entity "CommunalHelper/ResetStateCrystal" ResetCrystal(
    x::Integer,
    y::Integer,
    oneUse::Bool=false,
)

const placements = Ahorn.PlacementDict(
    "Reset State Crystal (Communal Helper)" => Ahorn.EntityPlacement(
        ResetCrystal,
    ),
)

const sprite = "objects/CommunalHelper/resetStateCrystal/ghostIdle00"

function Ahorn.selection(entity::ResetCrystal)
    x, y = Ahorn.position(entity)

    return Ahorn.getSpriteRectangle(sprite, x, y)
end

Ahorn.render(ctx::Ahorn.Cairo.CairoContext, entity::ResetCrystal) =
    Ahorn.drawSprite(ctx, sprite, 0, 0; tint=(0.35, 0.35, 0.35, 1.0))

end

File: Loenn/entities/reset_state_crystal.lua
local utils = require("utils")

local resetStateCrystal = {}

resetStateCrystal.name = "CommunalHelper/ResetStateCrystal"
resetStateCrystal.depth = -100
resetStateCrystal.placements = {
    {
        name = "reset_state_crystal",
        data = {
            oneUse = false
        }
    }
}

resetStateCrystal.texture = "objects/CommunalHelper/resetStateCrystal/ghostIdle00"
resetStateCrystal.color = {0.35, 0.35, 0.35, 1.0}

function resetStateCrystal.rectangle(room, entity)
    return utils.rectangle(entity.x - 5, entity.y - 5, 10, 10)
end

return resetStateCrystal
